### PR TITLE
Implement backends

### PR DESCRIPTION
**Pull request by jonasteuwen**
This draft PR implements different backends (openslide, pyvips and tifffile). This can be used as follows:

```
from dlup import SlideImage
from dlup.backends import ImageBackends

slide_openslide = SlideImage.from_file_path(path, backend=ImageBackends.OPENSLIDE)
slide_pyvips = SlideImage.from_file_path(path, backend=ImageBackends.PYVIPS)
slide_tifffile = SlideImage.from_file_path(path, backend=ImageBackends.TIFFFILE)

# Try to autodetect the backend
slide_autodetect = SlideImage.from_file_path(path, backend=ImageBackends.AUTODETECT)
```

### CODE REVIEW

File: dlup/experimental_backends/__init__.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        try:
+            return _try_pyvips(filename)
+        except UnsupportedSlideError:
+            pass
+        try:
+            return _try_tifffile(filename)
+        except UnsupportedSlideError:
+            raise UnsupportedSlideError(f"Cannot read {filename} with pyvips or tifffile.")
+
+    try:
```
Comment (lromor) [inline dlup/experimental_backends/__init__.py:52]: pyvips seems to already "try" to load via openslide. Maybe just use pyvips?

File: tests/test_backends.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import numpy as np
```
Comment (lromor) [inline tests/test_backends.py:1]: This shouldn't be included in the PR probably right?

**Review by lromor — COMMENTED** [review]
Only a few comments. We should device a different type of testing for checking backends. But let's see if it's worth it. Another thing is the redundant "openslide" backend. 
Other than that, great features!

File: dlup/experimental_backends/_tifffile.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    tile_per_line = int(np.ceil(image_width / tile_width))
+
+    out = np.empty(
+        (page.imagedepth, (tile_y1 - tile_y0) * tile_height, (tile_x1 - tile_x0) * tile_width, page.samplesperpixel),
+        dtype=page.dtype,
+    )
+
+    fh = page.parent.filehandle
+
+    jpeg_tables = page.tags.get("JPEGTables", None)
```
Comment (lromor) [inline dlup/experimental_backends/_tifffile.py:82]: Could you explain in layman terms what are you doing here? Just for future readers.

Comment (jonasteuwen) [reply dlup/experimental_backends/_tifffile.py:82]: I took it from the link in the comment, It's the author providing this for us.

File: tests/test_backends.py (hunk 2 above)
Comment (jonasteuwen) [reply tests/test_backends.py:1]: No.

File: dlup/experimental_backends/__init__.py (hunk 1 above)
Comment (jonasteuwen) [reply dlup/experimental_backends/__init__.py:52]: Yeah but it can depend on another version.
